src/main/java/mate/academy/springboot/datajpa/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.util.NoSuchElementException;
+import mate.academy.springboot.datajpa.dao.CategoryRepository;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public Category create(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category getById(Long id) {
+        return categoryRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("Can't find the category by "
+                        + "id: " + id));
+    }
+
+    @Override
+    public void delete(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public Category update(Category category, Long id) {
+        Category oldCategory = categoryRepository.findById(id)
```

Comment:
I prefer to use getById method,  Category oldCategory= this.getById(id);

Reply:
I don't know what would be better here, I thought that when updating an entity, I should check whether it is in the database with this ID and then update it.

Reply:
I think @ykhmaruk meant that you have code duplication here and exactly the same 'find' operation was implemented within your service's getById() method, so you can reuse it here

Reply:
Yes, thank you, I later figured out what he meant and really realized that it would be better, or at least avoid duplication of code